samza-core/src/main/java/org/apache/samza/clustermanager/SamzaResourceRequest.java

```diff
@@ -63,7 +69,11 @@
   private final Instant requestTimestamp;
 
   public SamzaResourceRequest(int numCores, int memoryMB, String preferredHost, String processorId) {
-    this(numCores, memoryMB, preferredHost, processorId, Instant.now());
+    this(numCores, memoryMB, preferredHost, processorId, Instant.now(), null);
```

Comment:
emptyset instead of null?
however, what does empty set/null mean? 
will the request ignore fault domain notions if its null but somehow break the behavior if its an empty set?

Reply:
When we make the actual container request to Yarn, the null value will ensure that the rack requirement for that request is ignored, similar to how it currently is without rack awareness. The code on Yarn side checks specifically for it to be null.

Reply:
@lakshmi-manasa-g it should still work as is.